azurerm/resource_arm_app_service_plan.go

```diff
@@ -0,0 +1,215 @@
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArmAppServicePlan() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAppServicePlanCreateUpdate,
+		Read:   resourceArmAppServicePlanRead,
+		Update: resourceArmAppServicePlanCreateUpdate,
+		Delete: resourceArmAppServicePlanDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"location": locationSchema(),
+			"sku": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"tier": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"size": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+				Set: resourceAzureRMAppServicePlanSkuHash,
+			},
+			"maximum_number_of_workers": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceArmAppServicePlanCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	AppServicePlanClient := client.appServicePlansClient
+
+	log.Printf("[INFO] preparing arguments for AzureRM App Service Plan creation.")
+
+	resGroup := d.Get("resource_group_name").(string)
+	name := d.Get("name").(string)
+	location := d.Get("location").(string)
+
+	sku := expandAzureRmAppServicePlanSku(d)
+
+	properties := web.AppServicePlanProperties{}
+	if v, ok := d.GetOk("maximum_number_of_workers"); ok {
+		maximumNumberOfWorkers := v.(int32)
+		properties.MaximumNumberOfWorkers = &maximumNumberOfWorkers
+	}
+
+	appServicePlan := web.AppServicePlan{
+		Location:                 &location,
+		AppServicePlanProperties: &properties,
+		Sku: &sku,
+	}
+
+	_, error := AppServicePlanClient.CreateOrUpdate(resGroup, name, appServicePlan, make(chan struct{}))
+	err := <-error
+	if err != nil {
+		return err
+	}
+
+	read, err := AppServicePlanClient.Get(resGroup, name)
+	if err != nil {
+		return err
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read AzureRM App Service Plan %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	log.Printf("[DEBUG] Waiting for App Service Plan (%s) to become available", name)
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"Accepted", "Updating"},
+		Target:  []string{"Succeeded"},
+		Refresh: appServicePlanStateRefreshFunc(client, resGroup, name),
+		Timeout: 10 * time.Minute,
+	}
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf("Error waiting for App Service Plan (%s) to become available: %s", name, err)
```

Comment:
_minor_ can we make this second formatting argument `%+v` rather than `%s` to show the full error?